### PR TITLE
Environment SPA improvements

### PR DESCRIPTION
**Pull request by vrushaliwaykole**
Issue: #7370 

Description:
  - Environment name uniqueness validation on SPA.
  - Showing config repo link on pipelines modal and env variable modal

### CODE REVIEW

**Comment by vrushaliwaykole** [comment]
Config repo link on pipelines modal - 

Screenshot 2019-12-10 at 1 32 40 PM

File: server/src/main/webapp/WEB-INF/rails/webpack/views/components/environment_variables/index.tsx [hunk 1]
```diff
@@ -20,20 +20,24 @@ import s from "underscore.string";
...
+  }
+
   view(vnode: m.Vnode<GroupedEnvironmentVariablesAttrs>): m.Children {
     return <div>
       <h4 data-test-id={`${s.slugify(vnode.attrs.title)}-title`}>{vnode.attrs.title}</h4>
       {
         vnode.attrs.environmentVariables.map((envVar) => {
-          return <EnvironmentVariableWidget environmentVariable={envVar} onRemove={vnode.attrs.onRemove}/>;
+          return this.renderEnvironmentVariableWidget(vnode, envVar);
         })
```
Comment (kritika-singh3) [inline server/src/main/webapp/WEB-INF/rails/webpack/views/components/environment_variables/index.tsx]: The Closure is not required as there is only one statement. Can you pls convert it.

File: server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/config_repo_link.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+interface Attrs {
+  dataTestId?: string;
+  configRepoId: string;
+}
+
+export class ConfigRepoLink extends MithrilViewComponent<Attrs> {
+  view(vnode: m.Vnode<Attrs>) {
+    return <span data-test-id={vnode.attrs.dataTestId} className={styles.configRepoLink}>
+                                      (CONFIG REPO:
```
Comment (kritika-singh3) [inline server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/config_repo_link.tsx]: All caps `CONFIG REPO` does not look nice to me. How about `Config Repo: `?

Comment (vrushaliwaykole) [reply server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/config_repo_link.tsx]: `Config Repository:`  sounds good, will change it.

**Comment by vrushaliwaykole** [comment]
Conifg repo link on edit Environment variable modal - 
Screenshot 2019-12-10 at 1 50 35 PM

**Comment by vrushaliwaykole** [comment]
Config repo link on edit pipelines modal -

Screenshot 2019-12-10 at 1 50 19 PM

Config repo link on edit environment variables modal -

Screenshot 2019-12-10 at 2 14 55 PM

**Pull request merged**